Fix Query Param Passing

svanherk opened the pull request:
- We _do_ want the load function to run each time the entity changes
- Using a helper function until query parameter parsing is fixed for performSirenAction

svanherk commented on components/d2l-hm-filter/d2l-hm-filter.js:
```diff
@@ -405,6 +405,26 @@ class D2LHypermediaFilter extends mixinBehaviors([D2L.PolymerBehaviors.Siren.Ent
+	* It does not grab the query parameters correctly, and duplicate parameters and fields should not be included
+	*/
+	_performSirenActionWithQueryParams(action) {
+		const url = new URL(action.href, window.location.origin);
```
I need to confirm that BSI will handle this for us in IE11, or add that

svanherk replied:
Confirmed that it does not, so removed all non-IE11-compatible things

AlexBedley commented on components/d2l-hm-filter/d2l-hm-filter.js:
```diff
@@ -405,6 +405,26 @@ class D2LHypermediaFilter extends mixinBehaviors([D2L.PolymerBehaviors.Siren.Ent
+
+		url.searchParams.forEach(function(value, key) {
+			if (!action.fields.filter(x => x.name === key)[0]) {
+				action.fields.push({ name: key, value: value, type: 'hidden' });
```
is `hidden` necessary?

AlexBedley replied:
`performSirenAction` takes a second parameter called `fields` that could also work. But I think you're more familiar with the internals of `performSirenAction` than I am so I leave it up to you

svanherk replied:
I totally stole this from @jmurray87.  Jamie, did you find `hidden` necessary, and was there any reasoning between choosing this or using the second parameter?

jmurray87 replied:
I added it because of this line https://github.com/Brightspace/polymer-siren-behaviors/blob/bed996403a40dbb92675224df4fa2dc25e58ba5f/store/siren-action-behavior.js#L150 If we're going to add the query params as fields manually, I didn't want the action behaviour library to try to grab from url

AlexBedley replied:
I know there is at least 1 usage of `performSirenAction` where we add fields (in the master teacher stuff). Just wanna make sure it works in tandem with that.

jmurray87 replied:
We aren't modifying that call, that one is on the enrollments domain that won't have to worry about existing url param state

AlexBedley commented on components/d2l-hm-filter/d2l-hm-filter.js:
```diff
@@ -405,6 +405,26 @@ class D2LHypermediaFilter extends mixinBehaviors([D2L.PolymerBehaviors.Siren.Ent
+		}
+
+		url.searchParams.forEach(function(value, key) {
+			if (!action.fields.filter(x => x.name === key)[0]) {
```
You can use Carol's ie11 find hack `_findInArray`

AlexBedley commented on components/d2l-hm-filter/d2l-hm-filter.js:
```diff
@@ -405,6 +405,26 @@ class D2LHypermediaFilter extends mixinBehaviors([D2L.PolymerBehaviors.Siren.Ent
+	* Specifically, the getSirenFields function is broken: https://github.com/Brightspace/polymer-siren-behaviors/blob/master/store/siren-action-behavior.js#L14
+	* It does not grab the query parameters correctly, and duplicate parameters and fields should not be included
+	*/
+	_performSirenActionWithQueryParams(action) {
```
I definitely want to see unit tests for this guy

svanherk replied:
We're also going to be adding this to the eval hub list and Jamie was working on unit tests.  I can sync up with him and we can add it in both places

AlexBedley commented on components/d2l-hm-filter/d2l-hm-filter.js:
```diff
@@ -106,7 +106,7 @@ class D2LHypermediaFilter extends mixinBehaviors([D2L.PolymerBehaviors.Siren.Ent
 
 	async _loadFilters(entity) {
-		if (!entity || (this._filters && this._filters.length)) {
+		if (!entity) {
```
What's the reason behind this change?

svanherk replied:
Originally, we were saying "if this isn't the initial load and we've already loaded filters, skip this and do nothing" as a perf gain.  Which completely breaks the whole idea of reloading the filters and their info and self links when the table changes or the href of eval hub.  We want to reload everything if our entity ever changes

svanherk commented:
Updated to include no non-compatible IE11 code (other than the siren helpers themselves, grr) since this is a shared component and added tests.  Ready for re-review!
